### DISCUSSION TITLE
Get full message from user chat

### DISCUSSION THREAD

**Discussion by Diego-Schmidt** [vasani-arpit/WBOT#239] [Q&A]
Hello, following the php webhook example file, Is there any way to get the "complete" text of a message from a user?
for example: if($message == '/hi') will only read and set $message as /hi , how can i get everything on the chat if "if($message == '/hi') and user send /hi "I'm testsubject number one" , how can i get the text = "I'm testsubject number one" to use if on the response?
Thanks again for your time and work.

**Comment by vasani-arpit** [comment]
I think what you are looking for is search string method. You can't write if condition for each and every messages but you can write on the messages which contains the word `testsubject` so consider writing if conditions that way. 
I think this will help https://stackoverflow.com/a/4366748/2983629

Also if you can do menu based conversation where user is presented with menu whenever program doesn't understand user's message then it will be easy for the user to answer with just 1, 2 or what ever the menu items are. 

if you are looking for more conversation based architecture then I suggest you to go with https://botman.io/ this will handle lot of things for you. You just need to connect this with wbot.

**Reply by Diego-Schmidt** [reply]
My goal is to get the text after a "/command" , like "/addtoagenda Date with dentist at 2 PM" , is there any way to get "Date with dentist at 2 PM" part and add it to variable to use it on a, for example "store to txt"   . If i try to use $message it only gives me the "/addtoagenda" part and nothing else.

**Reply by vasani-arpit** [reply]
> My goal is to get the text after a "/command" , like "/addtoagenda Date with dentist at 2 PM" 

Use string functions. [strpos](https://www.php.net/manual/en/function.strpos.php) to find the command location in the string. or you can split the string using [explode](https://www.php.net/manual/en/function.explode.php) function by passing your command as separator. 

Have you think about https://wit.ai/ ? it is free and it will do this command separation for you using a AI.

**Reply by Diego-Schmidt** [reply]
And how do i get the full string using the php webhook?

**Reply by vasani-arpit** [reply]
It is being sent in webhook. do one thing open https://webhook.site copy the URL (and check CORS Headers checkbox) from there and set it as a webhook in WBOT. now when ever a message comes all the things which WBOT is sending to your php site will be listed there in the request. 

In short this is what WBOT will send in the webhook 👇

```js
{
"text":"This is full message text",
"type":"message",
"user":"MOBILENUMBER@c.us"
}
```

**Reply by Diego-Schmidt** [reply]
And what do i do with that? hahaha, sorry to disturb you with my questions.
I was trying to test what $message contains using this response on the php webhook, as a kind of echo reply. But I'm just getting back the command, not the full string.
`if($message == '/text')
{
	$response = array(array('text' => $message, 'type' => 'text'));
    $response = json_encode($response);
    echo($response);
}`
If i send a private message to the bot with "/text some words here" , the bot is replying just the command "/text" , not the "some words here" part.
Thanks for keep replying to my noob questions.

**Reply by vasani-arpit** [reply]
1. Start WBOT
2. Open dev tools in whatsapp window 
3. Go to network tab so that you can watch the request and responses
4. Change the code in php side. remove/comment if conditions just do `var_dump($request_body);` and `var_dump($_POST)`
this will print what php is getting in which variable. using which you can write the remaining logic.
